### ISSUE TITLE
Ssl certificate error 

### ISSUE THREAD

**Issue by Amogha-ark** [nolar/kopf#995]
### Keywords

Ssl certificate error

### Problem

When i try to deploy my operator on k8 cluster, it’s throwing error like ssl certificate like clientconnectorcertificateerror(connectionkey(host='10.xxxxxx', port=443, is_ssl=true, ssl=none, proxy=none, proxy_auth=none, proxy_headers_hash=xxxxxxxxxx), sslcertverificationerror(1, '[ssl: certificate_verify_failed] certificate verify failed: unable to get issuer certificate (_ssl.c:1131)'))
Basically it’s trying to do GET request to kubernetes api and hitting with this error !!
Should I need to configure anything on kopf ?

**Comment by gengjie** [comment]
I just encountered the same issue, how to fix it? how to disable the verify on ssl?

**Comment by msotoperdomo** [comment]
did you find a fix for this? I'm seeing this issue after upgrading to python 3.13.x

**Comment by Amogha-ark** [comment]
I stopped using Kopf, now using kubernetes library to perform my activities.

**Comment by otetard** [comment]
> did you find a fix for this? I'm seeing this issue after upgrading to python 3.13.x

You can try this kind of hack to support “insecure” X.509 certificates:

```python
import dataclasses

@kopf.on.login()
async def login_fn(logger, **kwargs):
    """Override the default authentication mechanism to support invalid X.509 certificates.

    This will call for `login_via_client`, the default authentication method and will change the `insecure` value of
    the dataclass returned.

    """

    result = kopf.login_via_client(logger=logger)

    if result is not None:
        result = dataclasses.replace(result, insecure=True)

    return result
```

Python 3.13 increased the security of [`ssl.create_default_context`](https://docs.python.org/3/library/ssl.html#ssl.create_default_context):

![Image](https://github.com/user-attachments/assets/e650540f-6999-4848-b93e-037bac0734f9)
